### ISSUE TITLE
ModelConfig: 14 of 16 parameters are undocumented

### ISSUE THREAD

**Issue by MohamedEldushi** [dataxid/dataxid-python#79]
# [DX] Large undocumented public surface: 16 ModelConfig parameters and 3 top-level classes/callables missing from Quickstart and Beta Test documentation

## Category
**Developer Experience (DX) Feedback** — combined report covering (a) undocumented `ModelConfig` parameters and (b) undocumented top-level symbols in the `dataxid` namespace. Both are documentation gaps rather than runtime bugs, so I am filing them together.

## Environment
- **OS:** Windows 11, Python 3.11.5
- **dataxid:** 0.2.0

## Summary

The `dataxid` package exposes a larger public surface than what users are told about in the Beta Test Challenge PDF and the Quickstart. Specifically:

1. `dataxid.ModelConfig` has **16 tunable parameters**. The PDF's "Example Tests" section names only `model_size` and `encoder_mode`.
2. The top-level `dataxid` namespace includes **`synthesize_tables` (callable) and `Table` (class)** that are not mentioned anywhere in the Quickstart, but are clearly intended for multi-table / relational workflows.
3. At least two pairs of parameters are ambiguously related (`timeout` vs `time_limit_seconds`, `max_epochs` vs `early_stop_patience`) without guidance on when to tune which.

A new user following the 5-line Quickstart has no path to discover features like reproducibility (`seed`), privacy mode (`privacy_enabled`), or multi-table synthesis (`synthesize_tables`) without reading the source code.

## Steps to Reproduce (enumeration-only, no API calls)

```python
import inspect
import dataxid

# Public classes/callables/modules exposed at the top level
public = {
    name: type(getattr(dataxid, name)).__name__
    for name in dir(dataxid) if not name.startswith("_")
}
print("Top-level public symbols:", len(public))
for name, kind in sorted(public.items()):
    print(f"  {name:24s}  {kind}")

# ModelConfig parameters and their defaults
cfg = dataxid.ModelConfig()
print("\nModelConfig parameters:", len(cfg.to_dict()))
for k, v in cfg.to_dict().items():
    print(f"  {k:24s} = {v!r}")
```

### Part 1 — Top-level symbols (27 in total)

```
APIError, AuthenticationError, ConflictError, DataxidError, InvalidRequestError,
Model, ModelConfig, ModelNotReadyError, NotFoundError, QuotaExceededError,
RateLimitError, Table, TrainingError, TrainingTimeoutError,
api_key, base_url, client, disable_logging, enable_logging,
encoder, exceptions, pipeline, synthesize, synthesize_tables, training
```

Of these, the PDF only mentions:
- `dataxid.synthesize`
- `dataxid.Model.create`
- `model.generate`
- `model.delete`
- `dataxid.ModelConfig`

**Undocumented top-level public symbols include:**
- `synthesize_tables` — a callable, presumably the multi-table / relational counterpart of `synthesize`
- `Table` — a public class whose purpose is not described anywhere in the Quickstart
- `disable_logging` / `enable_logging` — logging control with no documented effect on output
- `api_key`, `base_url` — module-level attributes that users might reasonably try to set directly instead of via env vars

### Part 2 — `ModelConfig` has 16 parameters (default values shown)

| # | Parameter | Default | Documented in PDF? |
|---|---|---|:---:|
|  1 | `embedding_dim`        | 64      | ❌ |
|  2 | `model_size`           | "medium"| ✅ (mentioned) |
|  3 | `batch_size`           | 256     | ❌ |
|  4 | `max_epochs`           | 100     | ❌ |
|  5 | `early_stop_patience`  | 4       | ❌ |
|  6 | `val_split`            | 0.1     | ❌ |
|  7 | `privacy_enabled`      | False   | ❌ (headline feature!) |
|  8 | `privacy_noise`        | 0.1     | ❌ |
|  9 | `encoding_types`       | None    | ❌ |
| 10 | `accumulation_steps`   | 1       | ❌ |
| 11 | `learning_rate`        | None    | ❌ |
| 12 | `label_smoothing`      | 0.0     | ❌ |
| 13 | `embedding_dropout`    | 0.5     | ❌ |
| 14 | `time_limit_seconds`   | 0.0     | ❌ |
| 15 | `seed`                 | None    | ❌ (critical for reproducibility) |
| 16 | `timeout`              | 14400.0 | ❌ (a 4-hour default!) |

**14 of 16 parameters have no user-facing documentation.** In particular:
- `seed` is the standard knob for reproducibility in every major ML library. Users will look for it and won't find it without reading source.
- `privacy_enabled` + `privacy_noise` are the headline selling point of DataXID (per the "What is DataXID" section of the PDF), yet they are not mentioned as tunable in the Beta Test materials.
- `timeout=14400.0` (4 hours) is silently applied to every call. Users cannot be expected to know this default exists.

## Expected Behavior

The API Reference page should enumerate:

1. **Every public top-level symbol** — at minimum `synthesize`, `synthesize_tables`, `Model`, `ModelConfig`, `Table`, the full exception hierarchy, `enable_logging` / `disable_logging`.
2. **Every `ModelConfig` parameter** with: name, type, default, unit (where applicable), allowed range, and effect on training.
3. **Cross-links** between related parameters (e.g., "see also `time_limit_seconds`" on `timeout`).

A minimal version would be a single autogenerated table. A good version groups parameters by concern (capacity, regularization, privacy, runtime) and shows a tuning example for each.

## Impact

- **New users abandon the 5-line Quickstart feeling they've seen the whole API.** They miss `seed`, `privacy_enabled`, `synthesize_tables`, and `timeout` — all features they would actively want.
- **Privacy claim is undersold.** The PDF's introduction calls privacy a core architectural feature, but the code-level knob (`privacy_enabled`, `privacy_noise`) is invisible to a user following the official docs.
- **Reproducibility claim is undersold.** Without discovering `seed`, users cannot produce deterministic synthetic data — making DataXID hard to use in regulated / audited settings where reproducibility is a compliance requirement.
- **Tuning is impossible.** A user who gets poor synthesis quality has no guidance on which of the 14 undocumented knobs to touch.
- **Wasted onboarding time.** Every new user who wants to go beyond the Quickstart has to read the source. That friction compounds across the user base.

## Suggested Fix

1. **Document all 16 ModelConfig parameters** in the API Reference with a single table (type, default, unit, effect). Example minimal entry:

   | Parameter | Type | Default | Effect |
   |---|---|---|---|
   | `seed` | `int \| None` | `None` | Seeds all randomness (model init, shuffling, generation) for reproducible output. Pass an integer to lock behavior across runs. |
   | `timeout` | `float` | `14400.0` | Maximum wall-clock seconds the client will wait for training before raising `TrainingTimeoutError`. Default is 4 hours; most workloads complete in under a minute. |
   | `privacy_enabled` | `bool` | `False` | If `True`, adds calibrated noise (`privacy_noise`) to gradients during training for differential-privacy guarantees. |

2. **Add a "Public API" overview page** that enumerates every top-level symbol (callables, classes, exceptions) with a one-line description and a link to its detail page. This single page should answer "what can I import from `dataxid`?" in under 30 seconds.

3. **Add a dedicated `synthesize_tables` / `Table` Quickstart.** If multi-table relational synthesis is supported, it should have at minimum the same 5-line example treatment as `synthesize`.

4. **Add `ModelConfig` to the Quickstart.** The current 5-line example uses defaults for everything. A second 5-line example showing `ModelConfig(seed=42, privacy_enabled=True, max_epochs=200)` would double the usefulness of the Quickstart without doubling its length.

## Reproducibility Package

Evidence collected by a deterministic pytest suite with JSONL logging. Every run appends one record per test to `results.jsonl`; the 16-parameter `ModelConfig` default table above is a direct copy of the `params` field of one such record. The suite is attached to the companion issues (#47 and #48) and can be re-run by any maintainer with:

```bash
pytest tests/test_config_inspection.py -v
```

## Multi-Environment Confirmation + New Finding in 0.3.0

Re-run on Google Colab on 2026-04-21:

- **Platform:** Linux-6.6.113+ (Colab)
- **Python:** 3.12.13
- **dataxid:** **0.3.0** (newer than the Windows run)

### Confirmation

- `ModelConfig` still exposes ~15 parameters; only `model_size` and `encoder_mode` are mentioned in the PDF / Quickstart.
- Top-level `dataxid` namespace now exposes **32 public symbols** (up from 27 in 0.2.0), and the new additions are also undocumented.

### New undocumented surface introduced in 0.3.0

Version 0.3.0 introduced a new nested `Privacy` dataclass on `ModelConfig`:

```python
>>> dataxid.ModelConfig().privacy
Privacy(protect_rare=True, rare_strategy='mask', enabled=False, noise=0.1)
```

Four new tunable fields (`protect_rare`, `rare_strategy`, `enabled`, `noise`) replace the previous flat `privacy_enabled` / `privacy_noise` parameters. None of them are mentioned in the Quickstart or the Beta Test PDF. In particular:

- `rare_strategy='mask'` is a non-trivial default that shapes how the model handles rare categorical values — this should be documented.
- `protect_rare=True` is on by default with no explanation of what "rare" means or how it's detected.
- `enabled=False` quietly disables the headline privacy feature of the product. A user expecting privacy-preserving behavior by default would be surprised.

**Recommendation:** add a dedicated "Privacy" section to the docs covering each of the four fields, their defaults, and their effects.

## Bonus: Tested on multiple environments ✅

**Comment by BurakUlver** [comment]
The full list is here: https://docs.dataxid.com/docs/configuration
We keep the Quickstart to the happy path.
